### ISSUE TITLE
code: 'ERR_INALID_URL' Unhandled Rejection! This is a bug and should be reported

### ISSUE THREAD

**Issue by csthompsondo2** [raineorshine/npm-check-updates#1522]
- [] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)

---Upgrading C:\Users\Scott\Downloads\AjaxControlToolkit-master\AjaxControlToolkit-master\AjaxControlToolkit\package.json
[--------------------] 0/6 0%TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:371:5)
    at onParseError (node:internal/url:552:9)
    at new URL (node:internal/url:628:5)
    at new URL (node:internal/url:625:22)
    at So (C:\Users\Scott\AppData\Roaming\npm\node_modules\npm-check-updates\build\index.js:418:21782)
    at IA (C:\Users\Scott\AppData\Roaming\npm\node_modules\npm-check-updates\build\index.js:424:483)
    at IA (C:\Users\Scott\AppData\Roaming\npm\node_modules\npm-check-updates\build\index.js:424:669) {
  input: './',
  code: 'ERR_INVALID_URL'
}
Unhandled Rejection! This is a bug and should be reported: https://github.com/raineorshine/npm-check-updates/issues

## Steps to Reproduce

.ncurc:

```js

```

Dependencies:

```json

```

Steps:

**Comment by csthompsondo2** [comment]
ncu -u -r . --packageFile **/package.json

**Comment by csthompsondo2** [comment]
Hi,

I received a number of warnings from Visual Studio 2015 loading and then
building the latest ajax control toolkit which I just downloaded from the
github site.  In that solution there are over 500 package.json files.
Updating all those files to eliminate the deprecated and no longer
available packages is impractical of course.  The command line -r parameter
on the application stands for recursive.  So the command syntax was
provided by chatgpt.com as a method to update the over 500 package.json
files in an automated manner recursively.

Thank you,

Scott Thompson

**Comment by csthompsondo2** [comment]
As a follow-up, the command was executed from an admin cmd prompt from the
project's root folder (where the .sln file is).

Thank you,

Scott Thompson

---------- Forwarded message ---------
From: Scott Thompson ***@***.***>
Date: Sun, Jun 15, 2025 at 8:38 AM
Subject: Re: [raineorshine/npm-check-updates] code: 'ERR_INALID_URL'
Unhandled Rejection! This is a bug and should be reported (Issue #1522)
To: raineorshine/npm-check-updates <
***@***.***>
Cc: raineorshine/npm-check-updates ***@***.***>,
Author ***@***.***>


Hi,

I received a number of warnings from Visual Studio 2015 loading and then
building the latest ajax control toolkit which I just downloaded from the
github site.  In that solution there are over 500 package.json files.
Updating all those files to eliminate the deprecated and no longer
available packages is impractical of course.  The command line -r parameter
on the application stands for recursive.  So the command syntax was
provided by chatgpt.com as a method to update the over 500 package.json
files in an automated manner recursively.

Thank you,

Scott Thompson

**Comment by raineorshine** [comment]
When using npm-check-updates, `-r` standards for `--registry`, not recursive.

The best I can do is provide a more informative error message, but ultimately this is a shell scripting error, not a bug in npm-check-updates. Thanks.
